### ISSUE TITLE
Emptying service check fields does not reassert template values, check fails

### ISSUE THREAD

**Issue by btassite** [centreon/centreon-engine#68]
Centreon 2.7.7
Centreon Engine 1.5.1
Centreon Broker 2.11.5

Symptoms: host X has service check with description "Ping" defined with 3 fields: count, warning, critical. This service check uses service template "Ping". The service template has predefined values for the 3 fields. If the host's service check has explicit values defined and you delete these, the service template's values are not reasserted and the check returns a critical with 100% packet loss. If you explicitly change the 3 fields to contain the same values as the service template, the check returns OK.

Also, perhaps a separate isse: could not use massive change to change these 3 fields for several hosts at the same time, the fields retain their previous contents, had to change all 3 fields manually for each host in turn.

**Comment by bouda1** [comment]
Hi,

In the service configuration, if a custom macro value is empty, it is really considered as empty. If you want to restore the template value, there is on the right of the value a turning arrow.

Regards.

**Comment by btassite** [comment]
I'm not sure why this was closed, the issue persists.

The three fields mentioned refer to "Args" under "Service Check Options", not using custom macros. In the "Custom macros" section it says "Nothing here, use the "Add" button".

Let me rephrase the problem:

If you add a check command and a template using this check with some predefined argument values (e.g. number of pings, warn/critical level values for percentage of packets lost), you can add a large number of service checks using this template without filling in the argument fields for every service check separately (the fields will remain empty if you look in the service check's configuration, the template's values are used however). As long as these fields remain empty in the service check configuration, if you change the template all service checks created with that template will use the new template values. If you fill in values for one or more checks, those service checks will use those values and are no longer affected by template changes for those values. If you now empty the values for a service check, the expected result is that the values defined for the template should be used again, but they aren't, the service check fails. In other words, the fields are "empty" when you first create the service check and it works, filling in and deleting values there will still appear as "empty" but the check's behaviour becomes different.

So now when you want to revert back to depending on the template's values there is no other option than to delete the check completely and readd it using the template, which asides from extra work means losing historic graphs/data for it.

The separate issue of massive change not working for this kind of thing also persists. If either issue is fixed in a new release please let us know.

**Comment by ganoze** [comment]
Unfortunately we are not able to reproduce your issue. This might be caused by use not using the same exact scenario that you do.

Could you provide us a video of your scenario or screenshots of every step you do ?
